tools/gulp/tasks/examples.ts

```diff
@@ -0,0 +1,247 @@
+  name = name.replace(/[A-Z]/g, ' $&');
+  name = name.toLowerCase().trim();
+  return name.split(' ').join('-')
+}
+
+/**
+ * Parse the AST of a file and get metadata about it
+ * 
+ * @param {string} filename 
+ * @param {string} src 
+ * @returns {{ primary: any, seconds: any[] }} 
+ */
+function createMetas(filename: string, src: string): { primaryComponent: any, secondaryComponents: any[] } {
+  const sourceFile = ts.createSourceFile(
+    filename, src, ts.ScriptTarget.Latest, false, ts.ScriptKind.TS);
+
+  const metas: any[] = [];
+
+  const visit = (node: any): void => {
+    if (node.kind === ts.SyntaxKind.ClassDeclaration) {
+      const meta: any = {
+        component: node.name.text
+      };
+
+      let primary = false;
+      if(node.jsDoc && node.jsDoc.length) {
+        for(const doc of node.jsDoc) {
+          if(doc.tags && doc.tags.length) {
+            for(const tag of doc.tags) {
+              const tagValue = tag.comment;
+              const tagName = tag.tagName.text;
+              if(tagName === 'title') {
+                meta.title = tagValue;
+                meta.primary = true;
+              }
+            }
+          }
+        }
+      }
+
+      if (node.decorators && node.decorators.length) {
+        for (const decorator of node.decorators) {
+          if (decorator.expression.expression.text === 'Component') {
+            for (const arg of decorator.expression.arguments) {
+              for (const prop of arg.properties) {
+                const name = prop.name.text;
+                const value = prop.initializer.text;
+                meta[name] = value;
+              }
+            }
+          }
+        }
+      }
+
+      metas.push(meta);
+    }
+
+    ts.forEachChild(node, visit);
+  };
+
+  visit(sourceFile);
+
+  return {
+    primaryComponent: metas.find(m => m.primary),
+    secondaryComponents: metas.filter(m => !m.primary)
+  };
+}
+
+/**
+ * Creates the examples module and metadata
+ */
+task('build-examples-module', (cb) => {
+  const results: ExampleMetadata[] = [];
+
+  glob('src/material-examples/**/*.ts', (err, matches) => {
+    if (err) {
+      throw err;
+    }
+
+    for (const match of matches) {
+      const src = fs.readFileSync(match, 'utf-8');
+      const filename = match
+        .replace('src/material-examples/', './')
+        .replace('.ts', '');
+       
+      const { primaryComponent, secondaryComponents } = createMetas(match, src);
+
+      if(primaryComponent) {
+        // convert the class name to dashcase id
+        let id = primaryComponent.component.replace('Example', '');
+        id = convertToDashCase(id);
+        
+        const example: ExampleMetadata = {
+          filename,
+          id,
+          component: primaryComponent.component,
+          title: primaryComponent.title,
+          additionalComponents: [],
+          additionalFiles: [],
+          selectorName: []
+        };
+
+        if(secondaryComponents.length) {
```

Comment:
Space missing here and in the for loop below as well. (Do we have a lint rule for this?)